tests/test_5_protein_dl.py

```diff
@@ -0,0 +1,351 @@
+            # multiple exons
+            strand = cds_exons.iloc[0].Strand
+            assert np.all(strand == cds_exons.Strand)
+
+            intervals = [gtf_row2interval(row)
+                         for i, row in cds_exons.loc[transcript_id].sort_values("Start").iterrows()]
+        return intervals, strand
+
+    def get_seq(self, transcript_id):
+        exons, strand = self.get_cds_exons(transcript_id)
+        # merge the sequences
+
+        seq = "".join([self.fae.extract(exon) for exon in exons])
+
+        if strand == '-':
+            # optionally reverse complement
+            seq = rc_dna(seq)
+        while(len(seq) % 3 != 0):
+            seq = seq[:-1]
+        return seq
+
+    def get_seq_variants(self, transcript_id, variants):
+        exons, strand = self.get_cds_exons(transcript_id)
+        # merge the sequences
+
+        # TODO - insert genetic variants here
+        seq = "".join([self.fae.extract(exon) for exon in exons])
+
+        if strand == '-':
+            # optionally reverse complement
+            seq = rc_dna(seq)
+        return seq
+
+    def __getitem__(self, idx):
+        return self.get_seq(self.transcripts[idx])
+
+    def overlaped_exons(self, variant):
+        """Which exons are overlapped by a variant
+
+        Overall strategy:
+        1. given the variant, get all the affected transcripts
+        2. Given the transcript and the variants,
+          fetch the ref and alt sequences for the transcripts
+        """
+        # TODO - perform a join between variants and exons
+        # https://github.com/gagneurlab/MMSplice/blob/master/mmsplice/vcf_dataloader.py#L136-L190
+        # this will generate (variant, cds_exon) pairs
+        # cds_exon will contain also the information about the order in the transcript
+        return self.cds_exons.join(variant)
+
+
+class AminoAcidVCFSeqExtractor:
+
+    def __init__(self, fasta_file, vcf_file, gtf_file):
+        self.gtf_file = str(gtf_file)
+        self.fasta_file = str(fasta_file)
+        self.vcf_file = str(vcf_file)
+        self.gps = GenomeCDSSeq(self.gtf_file, self.fasta_file)
+        self.sse = SingleSeqVCFSeqExtractor(self.fasta_file, self.vcf_file)
+        self.sve = SingleVariantVCFSeqExtractor(self.fasta_file, self.vcf_file)
+    # if the dna seq is not %3==0, there are unnecessary bases at the end
+
+    # should be called only after all exons are connected!
+
+    def cut_seq(self, seq):
+        while(len(seq) % 3 != 0):
+            seq = seq[:-1]
+        return seq
+
+
+class SingleSeqAminoAcidVCFSeqExtractor(AminoAcidVCFSeqExtractor):
+
+    # function for a concrete protein id
+
+    def extract(self, transcript_id, anchor, sample_id=None):
+        intervals, strand = self.gps.get_cds_exons(transcript_id)
+        seq = ""
+        for interval in intervals:
+            interval = Interval(interval[0], int(interval[1]), int(interval[2]))
+            seq = "".join([seq, self.sse.extract(interval, anchor=anchor)])
```

Comment:
seq = ''.join([
  self.sse.extract(interval, anchor=anchor)]
   for interval in intervals
])